editor/editor_node.cpp

```diff
@@ -6864,6 +6872,13 @@ EditorNode::EditorNode() {
 		SurfaceUpgradeTool::get_singleton()->begin_upgrade();
 	}
 
+	// Same for UID upgrade tool.
+	uid_upgrade_tool = memnew(UIDUpgradeTool);
+	run_uid_upgrade_tool = EditorSettings::get_singleton()->get_project_metadata("uid_upgrade_tool", "run_on_restart", false);
+	if (run_uid_upgrade_tool) {
+		UIDUpgradeTool::get_singleton()->begin_upgrade();
+	}
```

Comment:
Note that UIDUpgradeTool is Object, so it will leak if you don't memdelete it.
I guess that's the reason to have `uid_upgrade_tool` variable? Although you can memdelete singleton as well, so not sure why it exists.

Reply:
I see I missed the `memdelete` for `surface_upgrade_tool` when I was working on this - I'll add it soon!